topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReplicasChangeClient.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    private void updateToPending(List<ReconcilableTopic> reconcilableTopics, String message) {
+        LOGGER.infoOp("{}, Topics: {}", message, topicNames(reconcilableTopics));
+        reconcilableTopics.forEach(reconcilableTopic ->
+            reconcilableTopic.kt().setStatus(new KafkaTopicStatusBuilder(reconcilableTopic.kt().getStatus())
+                .withReplicasChange(new ReplicasChangeStatusBuilder()
+                    .withState(PENDING).withTargetReplicas(reconcilableTopic.kt().getSpec().getReplicas()).build()).build()));
+    }
+    
+    private void updateToOngoing(List<ReconcilableTopic> reconcilableTopics, String message, String userTaskId) {
+        LOGGER.infoOp("{}, Topics: {}", message, topicNames(reconcilableTopics));
+        reconcilableTopics.forEach(reconcilableTopic ->
+            reconcilableTopic.kt().setStatus(new KafkaTopicStatusBuilder(reconcilableTopic.kt().getStatus())
+                .editOrNewReplicasChange().withState(ONGOING).withSessionId(userTaskId).endReplicasChange().build()));
+    }
+    
+    private void updateToCompleted(List<ReconcilableTopic> reconcilableTopics, String message) {
+        LOGGER.infoOp("{}, Topics: {}", message, topicNames(reconcilableTopics));
+        reconcilableTopics.forEach(reconcilableTopic ->
+            reconcilableTopic.kt().setStatus(new KafkaTopicStatusBuilder(reconcilableTopic.kt().getStatus())
+                .withReplicasChange(null).build()));
+    }
+    
+    private void updateToFailed(List<ReconcilableTopic> reconcilableTopics, String message, HttpResponse<String> response) {
+        Optional<String> error = parseErrorResponse(response);
+        String text = error.isPresent() ? format("%s, %s", message, error.get()) : message;
+        LOGGER.errorOp("{}, Topics: {}", text, topicNames(reconcilableTopics));
+        reconcilableTopics.forEach(reconcilableTopic ->
+            reconcilableTopic.kt().setStatus(new KafkaTopicStatusBuilder(reconcilableTopic.kt().getStatus())
+                .editOrNewReplicasChange().withMessage(text).endReplicasChange().build()));
+    }
+
+    private Optional<String> parseErrorResponse(HttpResponse<String> response) {
+        if (response != null && response.body() != null) {
+            try {
+                ErrorResponse errorResponse = mapper.readValue(response.body(), ErrorResponse.class);
+                if (errorResponse.errorMessage().contains("NotEnoughValidWindowsException")) {
+                    return Optional.of("Cluster model not ready");
+                } else if (errorResponse.errorMessage().contains("OngoingExecutionException")
+                    || errorResponse.errorMessage().contains("stop_ongoing_execution")) {
+                    return Optional.of("Another task is executing");
+                } else {
+                    return Optional.of(errorResponse.errorMessage());
+                }
+            } catch (Throwable t) {
+                // ignore
```

Comment:
We should at least handle the case of a deserialization exception by returning a message, I think.
We should at least log any other `Exception` types at `warn`.
We should log any other `Throwable` at `error`.